services/venice-server/src/main/java/com/linkedin/venice/listener/StatsHandler.java

```diff
@@ -305,58 +297,65 @@ private void recordBasicMetrics() {
-        currentStats.recordRequestFirstPartLatency(storeName, firstPartLatency);
+        serverHttpRequestStats.recordRequestFirstPartLatency(firstPartLatency);
       }
       if (partsInvokeDelayLatency > 0) {
-        currentStats.recordRequestPartsInvokeDelayLatency(storeName, partsInvokeDelayLatency);
+        serverHttpRequestStats.recordRequestPartsInvokeDelayLatency(partsInvokeDelayLatency);
       }
       if (secondPartLatency > 0) {
-        currentStats.recordRequestSecondPartLatency(storeName, secondPartLatency);
+        serverHttpRequestStats.recordRequestSecondPartLatency(secondPartLatency);
       }
       if (requestPartCount > 0) {
-        currentStats.recordRequestPartCount(storeName, requestPartCount);
+        serverHttpRequestStats.recordRequestPartCount(requestPartCount);
       }
       if (readComputeLatency >= 0) {
-        currentStats.recordReadComputeLatency(storeName, readComputeLatency, isAssembledMultiChunkLargeValue());
+        serverHttpRequestStats.recordReadComputeLatency(readComputeLatency, isAssembledMultiChunkLargeValue());
       }
       if (readComputeDeserializationLatency >= 0) {
-        currentStats.recordReadComputeDeserializationLatency(
-            storeName,
+        serverHttpRequestStats.recordReadComputeDeserializationLatency(
             readComputeDeserializationLatency,
             isAssembledMultiChunkLargeValue());
       }
       if (readComputeSerializationLatency >= 0) {
-        currentStats.recordReadComputeSerializationLatency(
-            storeName,
-            readComputeSerializationLatency,
-            isAssembledMultiChunkLargeValue());
+        serverHttpRequestStats
+            .recordReadComputeSerializationLatency(readComputeSerializationLatency, isAssembledMultiChunkLargeValue());
       }
       if (dotProductCount > 0) {
-        currentStats.recordDotProductCount(storeName, dotProductCount);
+        serverHttpRequestStats.recordDotProductCount(dotProductCount);
       }
       if (cosineSimilarityCount > 0) {
-        currentStats.recordCosineSimilarityCount(storeName, cosineSimilarityCount);
+        serverHttpRequestStats.recordCosineSimilarityCount(cosineSimilarityCount);
       }
       if (hadamardProductCount > 0) {
-        currentStats.recordHadamardProductCount(storeName, hadamardProductCount);
+        serverHttpRequestStats.recordHadamardProduct(hadamardProductCount);
       }
       if (countOperatorCount > 0) {
-        currentStats.recordCountOperatorCount(storeName, countOperatorCount);
+        serverHttpRequestStats.recordCountOperator(countOperatorCount);
       }
       if (isRequestTerminatedEarly) {
-        currentStats.recordEarlyTerminatedEarlyRequest(storeName);
+        serverHttpRequestStats.recordEarlyTerminatedEarlyRequest();
+      }
+      if (keySizeList != null) {
+        for (int i = 0; i < keySizeList.size(); i++) {
+          serverHttpRequestStats.recordKeySizeInByte(keySizeList.getInt(i));
```

Comment:
I think we should check that the item is > 0, and only then record it. Otherwise, we could be recording improper stuff in cases of missing values... more details here: https://github.com/linkedin/venice/pull/153#discussion_r1055991544

And same for the other loop below.

Reply:
I guess for value size that makes sense, but even for not-found keys, we should record them regardless ?

Reply:
Oh yeah, right... this check should only apply to value size.